tests/ui/generic-associated-types/bugs/issue-100013.stderr

```diff
@@ -7,16 +7,6 @@ LL | |         async {}.await; // a yield point
 LL | |     }
    | |_____^
    |
-note: the lifetime defined here...
-  --> $DIR/issue-100013.rs:16:38
-   |
-LL |         let x = None::<I::Future<'_, '_>>; // a type referencing GAT
-   |                                      ^^
-note: ...must outlive the lifetime defined here
-  --> $DIR/issue-100013.rs:16:34
-   |
-LL |         let x = None::<I::Future<'_, '_>>; // a type referencing GAT
-   |                                  ^^
```

Comment:
This is *the* reason that I added `Option<Span>` to `BrAnon`: because we "erase" all regions in generators and replace them wtih `BrAnon`s, and later try to prove lifetime bounds on them.

Without the span here, this is a very confusing diagnostic.

Reply:
Ugh, all for a single use :/

Reply:
Yeah, it kind of sucks.

I'm open to other ways to emit helpful clues to what's going on without requiring a `Span`. For example, I could imagine a more vague "some lifetime in this async block has an outlives bound as required here ..." (we'd still emit a note that it's a bug).

Another open might be to store the Spans in some side table for the generator and use the bound var index to access that, but I imagine that will take extra plumbing *elsewhere*.